packages/shared/src/components/squads/Details.tsx

```diff
@@ -13,39 +15,70 @@ import { formToJson } from '../../lib/form';
 import { Modal } from '../modals/common/Modal';
 import { blobToBase64 } from '../../lib/blob';
 import { checkExistingHandle, SquadForm } from '../../graphql/squads';
+import { capitalize } from '../../lib/strings';
 
 const squadImageId = 'squad_image_file';
+
+interface SquadDetailsProps {
+  onSubmit: (e: FormEvent, formJson: SquadForm) => void;
+  form: Partial<SquadForm>;
+  createMode: boolean;
+}
+
+const getFormData = async (
+  current: SquadForm,
+  imageChanged: boolean,
+): Promise<SquadForm> => {
+  if (!imageChanged) return current;
+
+  const input = document.getElementById(squadImageId) as HTMLInputElement;
+  const file = input.files[0];
+  const base64 = await blobToBase64(file);
+
+  return { ...current, file: base64 };
+};
+
 export function SquadDetails({
   onSubmit,
   form,
   createMode = true,
-}: {
-  onSubmit: (e, formJson) => void;
-  form: Partial<SquadForm>;
-  createMode: boolean;
-}): ReactElement {
+}: SquadDetailsProps): ReactElement {
   const { name, handle, description } = form;
   const [imageChanged, setImageChanged] = useState(false);
-  const [handleValid, setHandleValid] = useState(true);
+  const [handleHint, setHandleHint] = useState<string>(null);
   const [canSubmit, setCanSubmit] = useState(!!name && !!handle);
+  const { mutateAsync: onValidateHandle } = useMutation(checkExistingHandle, {
+    onError: (err) => {
+      const clientError = err as ClientError;
+      const message = clientError?.response?.errors?.[0]?.message;
+      if (!message) return null;
+
+      const error = JSON.parse(message);
+      if (error?.handle) return setHandleHint(capitalize(error?.handle));
+
+      const [unknown] = Object.values(error);
+
+      return setHandleHint(unknown as string);
+    },
+  });
   const handleSubmit = async (e: FormEvent<HTMLFormElement>) => {
     e.preventDefault();
     const formJson = formToJson<SquadForm>(e.currentTarget);
+
     if (!formJson.name || !formJson.handle) {
-      return;
-    }
-    if (imageChanged) {
-      const input = document.getElementById(squadImageId) as HTMLInputElement;
-      const file = input.files[0];
-      formJson.file = await blobToBase64(file);
-    }
-    const handleExists = createMode
-      ? await checkExistingHandle(formJson.handle)
-      : false;
-    setHandleValid(!handleExists);
-    if (!handleExists) {
-      onSubmit(e, formJson);
+      return null;
     }
+
+    const data = await getFormData(formJson, imageChanged);
+
+    if (!createMode) return onSubmit(e, data);
+
+    const handleExists = await onValidateHandle(formJson.handle);
+    setHandleHint(handleExists ? 'The handle is already exists' : null);
+
+    if (!handleExists) return onSubmit(e, data);
+
+    return null;
```

Comment:
I think this is unnecessary

Reply:
The linter is not accepting the function not having a return statement for having conditional returns.